nats.go

```diff
@@ -3511,8 +3543,12 @@ func (nc *Conn) close(status Status, doCBs bool) {
 
 	// Perform appropriate callback if needed for a disconnect.
 	if doCBs {
-		if nc.Opts.DisconnectedCB != nil && nc.conn != nil {
-			nc.ach.push(func() { nc.Opts.DisconnectedCB(nc) })
+		if nc.conn != nil {
+			if nc.Opts.DisconnectedErrCB != nil {
+				nc.ach.push(func() { nc.Opts.DisconnectedErrCB(nc, ErrConnectionClosed) })
```

Comment:
I wonder if maybe here we should use `nil` for the error instead of `ErrConnectionClosed`? That would mean that when calling `nc.Close()`, the disconnected callback is called but there is no error since it was called by the user.

Reply:
Right. We should then add this to the "doc" of DisconnectedErrCB that it is possible to get `nil` as an error, one case would be when the user closes the connection.

Reply:
Hi @wallyqs , thanks for review. Found possible missed case with err passing:
nats.go:1438 where we trying to initialize connected connection and got an err as result, if so we close connection with status `DISCONNECTED`. I think we should pass acquired error to new callback too, adding `err` as param to `close()` method? WDYT?
Adding err as param to `close()` will fix pointed issue when user closes connection, we will pass nil here with status `CLOSED`.

Reply:
I think that sounds good.